mods/ra/maps/infiltration/infiltration.lua

```diff
@@ -0,0 +1,343 @@
+	end
+end
+
+SovietBaseMaintenanceSetup = function()
+	local sovietbuildings = Utils.Where(Map.NamedActors, function(a)
+		return a.Owner == soviets
+			and a.HasProperty("StartBuildingRepairs") and a.HasProperty("Sell")
+	end)
+
+	-- This includes killed, captured (actor is temporarily removed) and sold.
+	Trigger.OnAllRemovedFromWorld(sovietbuildings, function()
+		Utils.Do(humans, function(player)
+			player.MarkCompletedObjective(destroyBase)
+		end)
+	end)
+
+	Utils.Do(sovietbuildings, function(sovietbuilding)
+		Trigger.OnDamaged(sovietbuilding, function(building)
+			if building.Health < building.MaxHealth * 3/4 then
+				building.StartBuildingRepairs()
+			end
+			if building.Health < building.MaxHealth * 1/4 then
+				building.Sell()
+			end
+		end)
+	end)
+end
+
+CheckPlayerDefeat = function()
+	if not reinforcementsHaveArrived then
+		return
+	end
+
+	Utils.Do(humans, function(player)
+		if player.HasNoRequiredUnits() then
+			player.MarkFailedObjective(destroyBase)
+		end
+	end)
+end
+
+labSecured = false
+CheckLabSecured = function()
+	if not reinforcementsHaveArrived or labSecured then
+		return
+	end
+
+	if player1.HasNoRequiredUnits() and player2 and player2.HasNoRequiredUnits() then
+		Utils.Do(humans, function(player)
+			player.MarkFailedObjective(secureLab)
+		end)
+	end
+
+	local radius = WDist.FromCells(10)
+	local labGuards = Utils.Where(Map.ActorsInCircle(Lab.CenterPosition, radius), function(a)
+		return a.Owner == soviets and a.HasProperty("Move")
+	end)
+
+	if #labGuards < 1 then
+		labSecured = true
+		Utils.Do(humans, function(player)
+			player.MarkCompletedObjective(secureLab)
+		end)
+		UserInterface.SetMissionText("")
+	end
+end
+
+Tick = function()
+	CapOre(soviets)
+	SecureLabTimer()
+	CheckLabSecured()
+	CheckPlayerDefeat()
+end
+
+WorldLoaded = function()
+	allies = Player.GetPlayer("Allies")
+	neutral = Player.GetPlayer("Neutral")
+	creeps = Player.GetPlayer("Creeps")
+	soviets = Player.GetPlayer("Soviets")
+
+	player1 = Player.GetPlayer("Allies1")
+	player2 = Player.GetPlayer("Allies2")
+	humans = { player1, player2 }
+
+	Utils.Do(humans, function(player)
+		if player and player.IsLocalPlayer then
+			Trigger.OnObjectiveAdded(player, function(p, id)
+				local objectiveType = string.lower(p.GetObjectiveType(id))
+				Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. objectiveType .. " objective")
+			end)
+
+			Trigger.OnObjectiveCompleted(player, function(p, id)
+				Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+			end)
+
+			Trigger.OnObjectiveFailed(player, function(p, id)
+				Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+			end)
+
+			Trigger.OnPlayerWon(player, function()
+				Media.PlaySpeechNotification(player, "MissionAccomplished")
+			end)
+
+			Trigger.OnPlayerLost(player, function()
+				Media.PlaySpeechNotification(player, "MissionFailed")
+			end)
+		end
+	end)
+
+	InsertSpies()
+	Utils.Do(TownAttackers, IdleHunt)
```

Comment:
Yes, I forgot to implement something like https://github.com/OpenRA/OpenRA/blob/1d9681a1d2769a9c736b79ba612204b979bf9d13/OpenRA.Mods.RA/Missions/Allies04Script.cs#L160

> Also, are they meant to destroy the hospital?

In the original, yes.